src/cy_serial_bridge/driver.py

```diff
@@ -364,6 +364,80 @@ def read_user_flash(self, addr: int, size: int) -> bytearray:
             result_bytes.extend(page_bytes)
 
         return result_bytes
+    
+
+    def is_gpio_output(self, gpio_nr: int) -> bool:
+        """
+        Check if a GPIO pin is in output mode.
+
+        :param pin: GPIO pin number to check
+        :return: True if the pin is an output pin, False otherwise
+        """
+
+        # TODO implement by checking it in the active configuration block
+        return True
+
+
+    def is_gpio_input(self, gpio_nr: int) -> bool:
+        """
+        Check if a GPIO pin is readable.
+        We can read the value of input or output pins, but
+        not if they are tristated or in some other mode.
+
+        :param pin: GPIO pin number to check
+        :return: True if the pin is an input pin, False otherwise
+        """
+
+        # TODO implement by checking it in the active configuration block
+        return True
+
+
+    def set_gpio(self, gpio_nr: int, value: int) -> None:
```

Comment:
Could the value be a bool here?

Reply:
As it turns out, yes. Initially I thought it might be a tri-state value, but it just returns 0 or 1